synapse/handlers/message.py

```diff
@@ -886,16 +886,19 @@ async def create_and_send_nonmember_event(
             )
 
             spam_check_result = await self.spam_checker.check_event_for_spam(event)
-            if isinstance(spam_check_result, Codes):
-                raise SynapseError(
-                    403,
-                    "This message had been rejected as probable spam",
-                    spam_check_result,
-                )
-            elif (
-                isinstance(spam_check_result, str)
-                and spam_check_result != self.spam_checker.NOT_SPAM
-            ):
+            if spam_check_result != self.spam_checker.NOT_SPAM:
+                # We need to check whether the return value is an error code
+                # specifically for backwards compatibility, since values in Codes
```

Comment:
this is confusing. I thought that the new way was to return an error code, but this implies that's only supported for backwards compat?